src/doc/rustc/src/platform-support/wasm32-unknown-emscripten.md

```diff
@@ -0,0 +1,116 @@
+If you are only targeting the web and need to access web APIs, the
+`wasm32-unknown-emscripten` target may be preferable.
+
+## Target maintainers
+
+- Hood Chatham, https://github.com/hoodmane
+- Sam Clegg, https://github.com/sbc100
+- Juniper Tyree, https://github.com/juntyr
+
+## Requirements
+
+This target is cross-compiled. The Emscripten compiler toolchain (emsdk) must be
+installed to link Rust-compiled code with Emscripten's libc and other C/C++
+code. Please refer to <https://emscripten.org/docs/getting_started/downloads.html>
+on how to get started with the emsdk.
+
+## Building the target
+
+Building this target can be done by:
+
+* Configure the `wasm32-unknown-emscripten` target to get built.
+* Configure LLD to be built.
+* Ensure the `WebAssembly` target backend is not disabled in LLVM.
+
+These are all controlled through `config.toml` options. It should be possible
+to build this target on any platform.
+
+## Building Rust programs
+
+Rust programs can be compiled by adding this target via rustup:
+
+```sh
+$ rustup target add wasm32-unknown-emscripten
+```
+
+and then compiling with the target:
+
+```sh
+$ rustc foo.rs --target wasm32-unknown-emscripten
+$ file foo.wasm
+```
+
+## Cross-compilation
+
+This target can be cross-compiled from any host.
+
+## Emscripten ABI Compatibility
+
+The Emscripten compiler toolchain does not follow a semantic versioning scheme
+that clearly indicates when breaking changes to the ABI can be made. If you are
+using the pre-compiled `std` from rustup, it may be compiled with a different
+Emscripten version than the local Emscripten you have installed. If a breaking
+change to the ABI was made in between the two versions, your code will exhibit
+undefined behaviour. If you come across issues, you can rebuild the Rust standard
+library with your local Emscripten version using
+
+```sh
+$ cargo +nightly -Zbuild-std build
+```
+
+## Testing
+
+This target is not extensively tested in CI for the rust-lang/rust repository.
+It's recommended to test the [`wasm32-wasip1`](./wasm32-wasip1.md) target instead
+for WebAssembly compatibility.
+
+## Conditionally compiling code
+
+It's recommended to conditionally compile code for this target with:
+
+```text
+#[cfg(all(target_family = "wasm", target_os = "emscripten"))]
+```
+
+## Enabled WebAssembly features
+
+WebAssembly is an evolving standard which adds new features such as new
+instructions over time. This target's default set of supported WebAssembly
+features will additionally change over time. The `wasm32-unknown-emscripten` target
+inherits the default settings of LLVM which typically, but not necessarily, matches
+the default settings of Emscripten as well.
+
+Please refer to the [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md)
+target's documentation on which WebAssembly features Rust enables by default, how
+features can be disabled, and how Rust code can be conditionally compiled based on
+which features are enabled.
```

Comment:
Over there, they list:
* `multivalue`
* `mutable-globals`
* `reference-types`
* `sign-ext`

These are all determined at compile time, so the advice over there should work. It does not apply as well to js/wasm interop features which are generally determined at link-time.

We might also note here that there is no support for `-fwasm-exceptions`, only for `-fexceptions` which will be added by default unless you compile with `panic=abort`. This is an important limitation. I would really like support for `wasm-exceptions`. 
https://github.com/rust-lang/rust/issues/112195

It would be nice to add wasm/js bigint interop (`-sWASM_BIGINT`) as another default feature. It's been around much longer than reference types and it makes a big improvement in code generation.

Reply:
I agree with enabling the `-sWASM_BIGINT` feature. This should be tracked in a separate PR.

Reply:
I've added a paragraph on the exception handling features.

Reply:
Opened a PR for `-sWASM_BIGINT`
https://github.com/rust-lang/rust/pull/131736